src/geometry/manifold/ManifoldGeometry.cc

```diff
@@ -162,8 +227,12 @@ std::shared_ptr<Polyhedron> ManifoldGeometry::toPolyhedron() const
 template std::shared_ptr<CGAL::Polyhedron_3<CGAL_Kernel3>> ManifoldGeometry::toPolyhedron() const;
 #endif
 
-std::shared_ptr<manifold::Manifold> binOp(const manifold::Manifold& lhs, const manifold::Manifold& rhs, manifold::OpType opType) {
-  return std::make_shared<manifold::Manifold>(lhs.Boolean(rhs, opType));
+ManifoldGeometry ManifoldGeometry::binOp(const ManifoldGeometry& lhs, const ManifoldGeometry& rhs, manifold::OpType opType) const {
+  auto mani = std::make_shared<manifold::Manifold>(lhs.manifold_->Boolean(*rhs.manifold_, opType));
```

Comment:
One thing to keep in mind: We should really be using `Manifold::BatchBoolean()`. I started on that refactoring but it hasn't made it yet. Shouldn't be too hard though, just loop-insert the children into the map.

Reply:
It would be great if that's the case - one less thing to worry about.